java.lang.NullPointerException after few seconds

Hi,
I tried to upgrade Maxwell to newest version 1.29.1 (from version 1.17.1), but I encountered a problem. After few seconds I'm getting a exception and process ends:

```
3478 [INFO] BinlogConnectorReplicator: Binlog disconnected.
3488 [INFO] TaskManager: Stopping 3 tasks
3488 [ERROR] TaskManager: cause:
java.lang.NullPointerException: null
        at com.zendesk.maxwell.replication.BinlogConnectorReplicator.processQueryEvent(BinlogConnectorReplicator.java:320) ~[maxwell-1.29.1.jar:1.29.1]
        at com.zendesk.maxwell.replication.BinlogConnectorReplicator.processQueryEvent(BinlogConnectorReplicator.java:338) ~[maxwell-1.29.1.jar:1.29.1]
        at com.zendesk.maxwell.replication.BinlogConnectorReplicator.getRow(BinlogConnectorReplicator.java:637) ~[maxwell-1.29.1.jar:1.29.1]
        at com.zendesk.maxwell.replication.BinlogConnectorReplicator.work(BinlogConnectorReplicator.java:178) ~[maxwell-1.29.1.jar:1.29.1]
        at com.zendesk.maxwell.util.RunLoopProcess.runLoop(RunLoopProcess.java:34) ~[maxwell-1.29.1.jar:1.29.1]
        at com.zendesk.maxwell.Maxwell.startInner(Maxwell.java:255) ~[maxwell-1.29.1.jar:1.29.1]
        at com.zendesk.maxwell.Maxwell.start(Maxwell.java:183) ~[maxwell-1.29.1.jar:1.29.1]
        at com.zendesk.maxwell.Maxwell.main(Maxwell.java:286) ~[maxwell-1.29.1.jar:1.29.1]
3498 [INFO] TaskManager: Stopping: com.zendesk.maxwell.schema.PositionStoreThread@4f07cf04
3501 [INFO] TaskManager: Stopping: com.zendesk.maxwell.bootstrap.BootstrapController@6ad1b8b3
3517 [INFO] TaskManager: Stopping: com.zendesk.maxwell.replication.BinlogConnectorReplicator@6a73d46c
java.lang.NullPointerException
        at com.zendesk.maxwell.replication.BinlogConnectorReplicator.processQueryEvent(BinlogConnectorReplicator.java:320)
        at com.zendesk.maxwell.replication.BinlogConnectorReplicator.processQueryEvent(BinlogConnectorReplicator.java:338)
        at com.zendesk.maxwell.replication.BinlogConnectorReplicator.getRow(BinlogConnectorReplicator.java:637)
        at com.zendesk.maxwell.replication.BinlogConnectorReplicator.work(BinlogConnectorReplicator.java:178)
        at com.zendesk.maxwell.util.RunLoopProcess.runLoop(RunLoopProcess.java:34)
        at com.zendesk.maxwell.Maxwell.startInner(Maxwell.java:255)
        at com.zendesk.maxwell.Maxwell.start(Maxwell.java:183)
        at com.zendesk.maxwell.Maxwell.main(Maxwell.java:286)
3836 [INFO] TaskManager: Stopped all tasks
```

I'm using command line option --replay , so maybe it is related to this exception.

I don't know anything about Java, but looking at the source code, couldn't be the problem that the called function setCurrentSchemaID has argument of type "long" instead of "Long"?

Could someone please fix this bug, please.
Stacktrace from version 1.33.0:
```
33631 [INFO] BinlogConnectorReplicator: Binlog disconnected.
33641 [INFO] TaskManager: Stopping 3 tasks
33642 [ERROR] TaskManager: cause:
java.lang.NullPointerException: null
        at com.zendesk.maxwell.replication.BinlogConnectorReplicator.processQueryEvent(BinlogConnectorReplicator.java:334) ~[maxwell-1.33.0.jar:1.33.0]
        at com.zendesk.maxwell.replication.BinlogConnectorReplicator.processQueryEvent(BinlogConnectorReplicator.java:352) ~[maxwell-1.33.0.jar:1.33.0]
        at com.zendesk.maxwell.replication.BinlogConnectorReplicator.getRow(BinlogConnectorReplicator.java:656) ~[maxwell-1.33.0.jar:1.33.0]
        at com.zendesk.maxwell.replication.BinlogConnectorReplicator.work(BinlogConnectorReplicator.java:180) ~[maxwell-1.33.0.jar:1.33.0]
        at com.zendesk.maxwell.util.RunLoopProcess.runLoop(RunLoopProcess.java:34) ~[maxwell-1.33.0.jar:1.33.0]
        at com.zendesk.maxwell.Maxwell.startInner(Maxwell.java:255) ~[maxwell-1.33.0.jar:1.33.0]
        at com.zendesk.maxwell.Maxwell.start(Maxwell.java:183) ~[maxwell-1.33.0.jar:1.33.0]
        at com.zendesk.maxwell.Maxwell.main(Maxwell.java:286) ~[maxwell-1.33.0.jar:1.33.0]
33644 [INFO] TaskManager: Stopping: com.zendesk.maxwell.schema.PositionStoreThread@10dcfdd4
33651 [INFO] TaskManager: Stopping: com.zendesk.maxwell.bootstrap.BootstrapController@7b5a81b
33658 [INFO] TaskManager: Stopping: com.zendesk.maxwell.replication.BinlogConnectorReplicator@50bb339
java.lang.NullPointerException
        at com.zendesk.maxwell.replication.BinlogConnectorReplicator.processQueryEvent(BinlogConnectorReplicator.java:334)
        at com.zendesk.maxwell.replication.BinlogConnectorReplicator.processQueryEvent(BinlogConnectorReplicator.java:352)
        at com.zendesk.maxwell.replication.BinlogConnectorReplicator.getRow(BinlogConnectorReplicator.java:656)
        at com.zendesk.maxwell.replication.BinlogConnectorReplicator.work(BinlogConnectorReplicator.java:180)
        at com.zendesk.maxwell.util.RunLoopProcess.runLoop(RunLoopProcess.java:34)
        at com.zendesk.maxwell.Maxwell.startInner(Maxwell.java:255)
        at com.zendesk.maxwell.Maxwell.start(Maxwell.java:183)
        at com.zendesk.maxwell.Maxwell.main(Maxwell.java:286)
33960 [INFO] TaskManager: Stopped all tasks
```

--- a/src/main/java/com/zendesk/maxwell/bootstrap/BootstrapController.java
+++ b/src/main/java/com/zendesk/maxwell/bootstrap/BootstrapController.java
@@ -89,7 +89,7 @@ public class BootstrapController extends RunLoopProcess  {
 		return this.currentSchemaID;
 	}
 
-	public synchronized void setCurrentSchemaID(long schemaID) {
+	public synchronized void setCurrentSchemaID(Long schemaID) {
 		this.currentSchemaID = schemaID;
 	}
 

--- a/src/test/java/com/zendesk/maxwell/MaxwellTestSupport.java
+++ b/src/test/java/com/zendesk/maxwell/MaxwellTestSupport.java
@@ -126,6 +126,28 @@ public class MaxwellTestSupport {
 		return new MaxwellContext(config);
 	}
 
+	public static MaxwellContext buildContext(Consumer<MaxwellConfig> maxwellConfigConsumer)
+			throws SQLException, URISyntaxException {
+		MaxwellConfig config = new MaxwellConfig();
+
+		config.replicationMysql.host = "127.0.0.1";
+		config.replicationMysql.user = "maxwell";
+		config.replicationMysql.password = "maxwell";
+		config.replicationMysql.sslMode = SSLMode.DISABLED;
+
+		config.maxwellMysql.host = "127.0.0.1";
+
+		config.maxwellMysql.user = "maxwell";
+		config.maxwellMysql.password = "maxwell";
+		config.maxwellMysql.sslMode = SSLMode.DISABLED;
+
+		config.databaseName = "maxwell";
+
+		maxwellConfigConsumer.accept(config);
+
+		return new MaxwellContext(config);
+	}
+
 	public static boolean inGtidMode() {
 		return System.getenv(MaxwellConfig.GTID_MODE_ENV) != null;
 	}

--- a/src/test/java/com/zendesk/maxwell/replication/BinlogConnectorReplicatorTest.java
+++ b/src/test/java/com/zendesk/maxwell/replication/BinlogConnectorReplicatorTest.java
@@ -11,11 +11,10 @@ import com.zendesk.maxwell.MaxwellContext;
 import com.zendesk.maxwell.MaxwellTestSupport;
 import com.zendesk.maxwell.MysqlIsolatedServer;
 import com.zendesk.maxwell.TestWithNameLogging;
-import com.zendesk.maxwell.bootstrap.SynchronousBootstrapper;
 import com.zendesk.maxwell.monitoring.NoOpMetrics;
 import com.zendesk.maxwell.producer.BufferedProducer;
 import com.zendesk.maxwell.producer.MaxwellOutputConfig;
-import com.zendesk.maxwell.row.RowMap;
+import com.zendesk.maxwell.producer.StdoutProducer;
 import com.zendesk.maxwell.schema.MysqlSchemaStore;
 import org.junit.Test;
 
@@ -131,5 +130,51 @@ public class BinlogConnectorReplicatorTest extends TestWithNameLogging {
 		assertEquals(222L, replicator.getRow().getData().get("i"));
 		assertEquals(333L, replicator.getRow().getData().get("i"));
 		assertEquals(null, replicator.getRow());
+	}
+
+	@Test
+	public void testSetNullSchemaIdInProcessQueryEvent() throws Exception {
+		assumeTrue(MysqlIsolatedServer.getVersion().atLeast(MysqlIsolatedServer.VERSION_5_6));
+
+		MysqlIsolatedServer server = MaxwellTestSupport.setupServer("--gtid_mode=ON --enforce-gtid-consistency=true");
+		MaxwellTestSupport.setupSchema(server, false);
+
+		server.execute("create table test.t ( i int )");
+		server.execute("create table test.u ( i int )");
+
+		Position position = Position.capture(server.getConnection(), true);
+
+		MaxwellContext context = MaxwellTestSupport.buildContext(config -> {
+			config.replicationMysql.port = server.getPort();
+			config.maxwellMysql.port = server.getPort();
+			config.filter = null;
+			config.initPosition = position;
+			config.replayMode = true;
+			config.producerType = "stdout";
+
+		});
+
+		BinlogConnectorReplicator replicator = new BinlogConnectorReplicator(
+				new MysqlSchemaStore(context, position),
+				new StdoutProducer(context),
+				context.getBootstrapController(null),
+				context.getConfig().maxwellMysql,
+				333098L,
+				"maxwell",
+				new NoOpMetrics(),
+				position,
+				false,
+				"maxwell-client",
+				new HeartbeatNotifier(),
+				null,
+				context.getFilter(),
+				new MaxwellOutputConfig(),
+				context.getConfig().bufferMemoryUsage
+		);
+
+		replicator.startReplicator();
+		server.execute("DROP TABLE IF EXISTS `xxx_tmp`");
+		replicator.getRow();
+
 	}
 }

